x-pack/plugins/fleet/server/constants/fleet_es_assets.ts

```diff
@@ -22,6 +28,22 @@ export const FLEET_GLOBAL_COMPONENT_TEMPLATE_CONTENT = {
       },
     },
     mappings: {
+      // All the dynamic field mappings
+      dynamic_templates: [
+        // This makes sure all mappings are keywords by default
+        {
+          strings_as_keyword: {
+            mapping: {
+              ignore_above: 1024,
+              type: 'keyword',
+            },
+            match_mapping_type: 'string',
+          },
+        },
+      ],
+      // As we define fields ahead, we don't need any automatic field detection
+      // This makes sure all the fields are mapped to keyword by default to prevent mapping conflicts
+      date_detection: false,
```

Comment:
I think we may need to use a separate component template for these global settings. We currently only include this component template when agent id verification is enabled: https://github.com/elastic/kibana/blob/990d069da13ea953f91a261299881f110809732c/x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts#L87-L90

I think it would make sense to name these a more clearly as well so we don't miss things like this in the future. It's too easy to assume that the "global" template is actually global. Maybe we have two separate ones:
- `.fleet_agent_id_verification-1`
- `.fleet_globals-1`